cmd/podman/rmi.go

```diff
@@ -53,12 +53,24 @@ func rmiCmd(c *cli.Context) error {
 
 	images := args[:]
 	var lastError error
+	var imagesToCheck []*image.Image
 	var imagesToDelete []*image.Image
 	if removeAll {
-		imagesToDelete, err = runtime.ImageRuntime().GetImages()
+		imagesToCheck, err = runtime.ImageRuntime().GetImages()
 		if err != nil {
 			return errors.Wrapf(err, "unable to query local images")
 		}
+		// Look up the image by it's repository rather than its ID.
+		// We need this later if we need to remove for Storage (Buildah).
+		for _, img := range imagesToCheck {
+			for repo := range image.ReposToMap(img.Names()) {
```

Comment:
@TomSweeneyRedHat im confused by this.  I'll catch you on IRC for follow-up

Reply:
@baude and I worked through this before scrum.  The issue I was running into was deleting container from an image created by Buildah when the image had a tag added to it.  The code as I had it here worked around that, however @baude pointed to a tweak of a if statement down lower that let me remove this code and still get the containers removed.  This has all been removed.